docs/os-building.md

```diff
@@ -0,0 +1,186 @@
+        initialized();
+        receiver = _receiver;
+    }
+
+    function setReceiver(address _newReceiver) auth(SET_RECEIVER_ROLE) public {
+        receiver = _receiver;
+    }
+
+    function sendFunds() payable isInitialized {
+        receiver.transfer(msg.value);
+    }
+}
+```
+
+An important note is that the `auth` modifier will also check whether the app is initialized. If the **app hasn't been initialized, the authentication check will fail**.
+
+When adding a role to your app, it needs to be added to the [`arapp.json`](cli-usage.md#the-arappjson-file) file, with a description of what having the role allows an entity to do. You can check Aragon's [Voting app arapp.json](https://github.com/aragon/aragon-apps/blob/master/apps/voting/arapp.json) for an example of role descriptions.
+
+```json
+{
+  ...,
+  roles: [
+    {
+      "name": "Set the receiver of funds",
+      "id": "SET_RECEIVER_ROLE",
+      "params": []
+    }
+  ]
+}
+```
+
+## Forwarding
+
+aragonOS introduces the concept of [forwarding](forwarding-intro.md) which is a generic interface for apps sending an intent to other apps if some conditions met. For example, a Voting app can be a forwarder that only forwards the intent if a vote passes, or a Payroll app that forwards intents to all employees of an organization.
+
+Forwarders are passed an EVMScript that can be executed by the app. For the time being, the script executed is always an array of calls that will be performed one after the other.
+
+In order for an app to be a forwarder it needs to implement these 3 functions in the `IForwarder` interface:
+
+```solidity
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/common/IForwarder.sol";
+
+contract MyApp is IForwarder, AragonApp {
+    address receiver;
+
+    function initialize(address _receiver) onlyInit public {
+        initialized();
+        receiver = _receiver;
+    }
+
+    function isForwarder() public pure returns (bool) {
+        return true;
+    }
+
+    function canForward(address _sender, bytes _evmCallScript) public view returns (bool) {
+        // arbitrary logic for deciding whether to forward a given intent
+        return _sender == receiver;
+    }
+
+    function forward(bytes _evmScript) public {
+        require(canForward(msg.sender, _evmScript));
+
+        bytes memory input = new bytes(0);
+        // an array of addresses that cannot be called from the script
+        address[] memory blacklist = new address[](0);
+
+        // Either immediately run the script or save it for later execution
+        runScript(_evmScript, input, blacklist); // actually executes script
+    }
+}
+```
+
+In the example above, this app will always forward an intent if the sender happens to be the `receiver` in the contract.
+
+An important note is that when a script is run, it will be able to perform calls from the app to addresses or contracts that are not in the address blacklist.
+
+## Fund recovery
```

Comment:
We should note here that depositing (into logic or base contracts) is also forbidden unless the depositable flag is set.